js/build.js

```diff
@@ -20,22 +19,36 @@ Fliplet.Widget.instance('image-gallery', function(data) {
       onResize: function() {
         wall.fitWidth();
         wall.refresh();
+      },
+      onComplete: function () {
+        $container.addClass('freewall-ready');
       }
     });
 
     if (!Fliplet.Env.get('interact')) {
       $(WALL_SELECTOR + ' .brick img').click(function() {
         var $clickedBrick = $(this)[0].parentElement;
-        data.options = data.options || {}
-        data.options.index = $clickedBrick.index - 1
 
-        Fliplet.Navigate.previewImages(data);
+        data.options = data.options || {};
+        data.options.index = $clickedBrick.index - 1;
+
+        // Update remote image URLs to authenticated URLs
+        Promise.all(_.map(_.get(data, 'images'), function (image, i) {
+          if (!Fliplet.Media.isRemoteUrl(image.url)) {
+            return Promise.resolve();
+          }
+
+          return Fliplet().then(function () {
+            _.set(data, ['images', i, 'url'], Fliplet.Media.authenticate(image.url));
```

Comment:
let's rename `i` to `idx` or `index` for clarity

Reply:
Done